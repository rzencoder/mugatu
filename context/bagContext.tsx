import { useState, useContext, createContext } from 'react'

const bagContext = createContext({})

export function ProvideBag({ children }: { children: any }): JSX.Element {
  const bag = useProvideBag()
  return <bagContext.Provider value={bag}>{children}</bagContext.Provider>
}

export const useBag = (): any => {
  return useContext(bagContext)
}

const useProvideBag = () => {
  const [bag, setBag] = useState([])

  const addToBag = (item: any) => {
    const newBag = [...bag, item]
    setBag(newBag)
  }

  const removeFromBag = (item) => {
    const updatedBag = bag.filter((el) => el.id !== item.id)
    setBag(updatedBag)
  }

  return {
    bag,
    addToBag,
    removeFromBag,
  } as const
}
