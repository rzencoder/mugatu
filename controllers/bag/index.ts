export { default as addToBag } from './addToBag'
export { default as getBag } from './getBag'
export { default as removeFromBag } from './removeFromBag'
export { default as getBagOnSignIn } from './getBagOnSignIn'
