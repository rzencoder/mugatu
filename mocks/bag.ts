export const bag = [
  {
    id: 'abc',
    name: 'Blue jeans',
    slug: 'blue-jeans',
    price: 29.99,
    rrp: 39.99,
    colour: 'blue',
    category: 'jeans',
    popular: true,
    gender: 'female',
    sizes: [
      { size: '6', stock: 10 },
      { size: '8', stock: 2 },
      { size: '10', stock: 0 },
    ],
    image: {
      url: '123',
    },
    quantity: 1,
    selectedSize: '6',
  },
  {
    id: 'def',
    name: 'Black Shirt',
    slug: 'black-shirt',
    price: 19.99,
    rrp: 24.99,
    colour: 'black',
    category: 'shirt',
    popular: false,
    gender: 'male',
    sizes: [
      { size: 's', stock: 10 },
      { size: 'm', stock: 2 },
      { size: 'l', stock: 0 },
    ],
    image: {
      url: '456',
    },
    quantity: 1,
    selectedSize: 's',
  },
]
