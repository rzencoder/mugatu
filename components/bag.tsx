import { useBag } from '@/context/bagContext'
import { useWishlist } from '@/context/wishlistContext'
import { DeleteIcon, ChevronDownIcon, InfoIcon } from '@chakra-ui/icons'
import {
  Flex,
  Box,
  Link,
  Button,
  useColorMode,
  Menu,
  MenuItem,
  MenuList,
  MenuButton,
} from '@chakra-ui/react'
import Image from 'next/image'
import NextLink from 'next/link'
import { useState, useEffect } from 'react'

const deliveryOptions = [
  {
    name: 'standard delivery',
    price: 4.95,
  },
  {
    name: 'next day delivery',
    price: 6.95,
  },
  {
    name: 'click and collect standard',
    price: 4.95,
  },
  {
    name: 'click and collect next day',
    price: 6.95,
  },
]

const Bag = () => {
  const { bag, updateBag, removeFromBag } = useBag()
  const { addToWishlist } = useWishlist()
  const { colorMode } = useColorMode()
  const [delivery, setDelivery] = useState(deliveryOptions[0])

  const getQuantityOptions = (item) => {
    const options = []
    const size = item.sizes.find((el) => el.size === item.selectedSize)
    for (let i = 1; i <= size.stock; i++) {
      if (i > 10) break
      options.push(i)
    }
    return options
  }

  const calculateSubTotal = () => {
    const total = bag.reduce((acc, cur) => {
      return cur.price * cur.quantity + acc
    }, 0)
    return total.toFixed(2)
  }

  const subtotal = calculateSubTotal()

  const displayDeliveryPrice = (price) => {
    if (price === 0) {
      return 'free'
    } else {
      return `£${price}`
    }
  }

  useEffect(() => {
    if (parseFloat(subtotal) > 50) {
      if (delivery.name === 'standard delivery' || delivery.name === 'click and collect standard') {
        if (delivery.price !== 0) {
          setDelivery({ ...delivery, price: 0 })
        }
      }
    } else {
      if (delivery.price === 0) {
        const originalPrice = deliveryOptions.find((option) => option.name === delivery.name)
        setDelivery({ ...delivery, price: originalPrice.price })
      }
    }
  }, [bag, delivery])

  if (!bag || bag.length === 0) {
    return (
      <Flex direction="column" justify="center" alignItems="center" minHeight="450px">
        <Box
          textTransform="lowercase"
          fontSize="28px"
          fontWeight="600"
          width="80%"
          textAlign="center"
          padding="0 0 30px"
        >
          Your Shopping Bag Is Empty
        </Box>
        <NextLink href="/" passHref>
          <Link>
            <Button>Continue Shopping</Button>
          </Link>
        </NextLink>
      </Flex>
    )
  }

  return (
    <Box bg={colorMode === 'light' ? '#eee' : 'mainBlack'}>
      <Flex
        direction={['column', null, 'row']}
        maxWidth="1000px"
        minHeight="500px"
        margin="0 auto"
        padding={['15px 5px', '20px']}
        justifyContent="space-between"
        position="relative"
      >
        <Flex width={['100%', null, '60%']} direction="column" mr={['5px', '20px']}>
          <Flex
            direction="column"
            p={['15px 10px', '20px']}
            mb="20px"
            bg={colorMode === 'light' ? 'mainWhite' : '#111'}
          >
            <Box p="0 0 15px" fontWeight="600" fontSize="28px">
              my bag
            </Box>
            {bag.map((item) => {
              return (
                <Flex key={`bag-item-${item.id}`} p="10px 0">
                  <Flex
                    minWidth={['120px', '140px']}
                    maxWidth={['120px', '140px']}
                    maxHeight={['180px', 'none']}
                  >
                    <Image src={item.image.url} width={150} height={225}></Image>
                  </Flex>
                  <Flex
                    direction="column"
                    p={['0 0 0 10px', '0 10px 0 15px']}
                    justifyContent="space-between"
                    width="100%"
                  >
                    <Flex justifyContent="space-between">
                      <NextLink
                        href={`/${item.gender === 'female' ? 'women' : 'men'}/catalog/${item.slug}`}
                        passHref
                      >
                        <Link>
                          <Box fontSize={['16px', '18px']} fontWeight="600">
                            {item.name}
                          </Box>
                        </Link>
                      </NextLink>
                      <Button
                        variant="icon"
                        fontSize="22px"
                        padding="5px"
                        margin={['0 3px', '0 10px']}
                        onClick={() => removeFromBag(item)}
                        color={colorMode === 'light' ? '#999' : '#aaa'}
                      >
                        <DeleteIcon
                          _hover={{
                            color: colorMode === 'light' ? '#777' : '#ccc',
                          }}
                        />
                      </Button>
                    </Flex>
                    <Flex p="5px 0" fontSize={['14px', '15px']} direction="column">
                      <Box p="0">{item.colour}</Box>
                      <Box p="0">size: {item.selectedSize}</Box>
                      <Flex alignItems="center">
                        <Box>qty:</Box>
                        <Menu>
                          <MenuButton
                            fontSize="15px"
                            height="20px"
                            as={Button}
                            rightIcon={<ChevronDownIcon m="0" />}
                            variant="transparentBg"
                            textTransform="lowercase"
                          >
                            {item.quantity}
                          </MenuButton>
                          <MenuList
                            bg={colorMode === 'light' ? 'mainWhite' : 'mainBlack'}
                            fontSize="16px"
                            minWidth="0"
                          >
                            {getQuantityOptions(item).map((el) => {
                              return (
                                <MenuItem
                                  key={`product-quantity-${el}`}
                                  onClick={() => updateBag({ ...item, quantity: el })}
                                >
                                  {el}
                                </MenuItem>
                              )
                            })}
                          </MenuList>
                        </Menu>
                      </Flex>
                    </Flex>
                    <Flex p="5px 0">
                      <Box fontSize={['19px', '22px']} color="mainRed">
                        £{item.price}
                      </Box>
                      <Box
                        fontSize={['14px', '15px']}
                        textDecoration="line-through"
                        margin="3px 0 0 8px"
                        color={colorMode === 'light' ? '#333' : '#ddd'}
                      >
                        £{item.rrp}
                      </Box>
                    </Flex>
                    <Button
                      fontSize={['14px', '15px']}
                      borderWidth="1px"
                      width={['150px', '170px']}
                      height={['35px', '45px']}
                      p={['1px', '3px 5px']}
                      _hover={{ bg: colorMode === 'light' ? 'mainBlackHover' : 'mainWhiteHover' }}
                      onClick={() => {
                        addToWishlist(item)
                        removeFromBag(item)
                      }}
                    >
                      <Box
                        backgroundImage="url(/icons/heart.png)"
                        width={['15px', '20px']}
                        height={['15px', '20px']}
                        backgroundSize="cover"
                        filter={colorMode === 'light' ? 'invert()' : 'none'}
                      />
                      <Box p="0 0 2px 5px" textTransform="lowercase">
                        Move to Wishlist
                      </Box>
                    </Button>
                  </Flex>
                </Flex>
              )
            })}
          </Flex>
          <Flex
            display={['none', null, 'flex']}
            justifyContent="space-between"
            bg={colorMode === 'light' ? 'mainWhite' : '#111'}
            p="20px"
            fontSize="22px"
            fontWeight="600"
          >
            <Box>sub-total</Box>
            <Box>£{subtotal}</Box>
          </Flex>
        </Flex>
        <Flex
          width={['100%', null, '40%']}
          direction="column"
          height={['350px']}
          bg={colorMode === 'light' ? 'mainWhite' : '#111'}
          p="20px"
          justifyContent="space-between"
        >
          <Box p="0 0 15px" fontWeight="600" fontSize="28px">
            summary
          </Box>
          <Flex direction="column" fontSize="16px">
            <Flex justifyContent="space-between" fontWeight="600">
              <Box>sub-total</Box>
              <Box>£{subtotal}</Box>
            </Flex>
            {subtotal < 50 && (
              <Flex
                bg={colorMode === 'light' ? 'mainBlack' : 'mainWhite'}
                color={colorMode === 'light' ? 'mainWhite' : 'mainBlack'}
                p="8px 12px"
                alignItems="center"
                margin="10px 0"
                fontWeight="600"
              >
                <InfoIcon fontSize="22px" mr="15px" />
                <Box>{`spend £${(50 - parseFloat(subtotal)).toFixed(
                  2
                )} more to qualify for free delivery`}</Box>
              </Flex>
            )}
            <Flex direction="column">
              <Flex justifyContent="space-between" fontWeight="600">
                <Box>delivery</Box>
                <Box>{displayDeliveryPrice(delivery.price)}</Box>
              </Flex>
              <Menu>
                <MenuButton
                  textAlign="left"
                  as={Button}
                  border="none"
                  textTransform="lowercase"
                  p="0"
                  fontWeight="400"
                  fontSize="16px"
                >
                  <Box as="span">{`${delivery.name} (${displayDeliveryPrice(
                    delivery.price
                  )})`}</Box>
                  <ChevronDownIcon />
                </MenuButton>
                <MenuList>
                  {deliveryOptions.map((option, index) => {
                    const price = option.name === delivery.name ? delivery.price : option.price
                    return (
                      <MenuItem
                        key={`delivery-option-${option.name}`}
                        onClick={() => setDelivery(deliveryOptions[index])}
                      >
                        {`${option.name} (${displayDeliveryPrice(price)})`}
                      </MenuItem>
                    )
                  })}
                </MenuList>
              </Menu>
            </Flex>
            <Flex justifyContent="space-between" fontSize="22px" fontWeight="600" p="5px 0">
              <Box>total</Box>
              <Box>£{(parseFloat(subtotal) + delivery.price).toFixed(2)}</Box>
            </Flex>
          </Flex>
          <NextLink href="/checkout" passHref>
            <Link>
              <Button
                bg="#0d6d33"
                border="none"
                color="mainWhite"
                width="100%"
                _hover={{
                  bg: '#15ab51',
                }}
              >
                Checkout
              </Button>
            </Link>
          </NextLink>
        </Flex>
      </Flex>
    </Box>
  )
}

export default Bag
