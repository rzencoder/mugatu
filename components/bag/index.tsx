import { useBag } from '@/context/bagContext'
import { calculateSubTotal } from '@/utils/bag'
import { Flex, Box, useColorMode } from '@chakra-ui/react'
import BagItem from './bagItem'
import EmptyBag from './emptyBag'
import Summary from './summary'

const Bag = () => {
  const { bag } = useBag()
  const { colorMode } = useColorMode()
  const subtotal = calculateSubTotal(bag)

  if (!bag || bag.length === 0) {
    return <EmptyBag />
  }

  return (
    <Box bg={colorMode === 'light' ? '#eee' : 'mainBlack'}>
      <Flex
        direction={['column', null, 'row']}
        maxWidth="1000px"
        minHeight="500px"
        margin="0 auto"
        padding={['15px 5px', '20px']}
        justifyContent="space-between"
        position="relative"
      >
        <Flex width={['100%', null, '60%']} direction="column" mr={['5px', '20px']}>
          <Flex
            direction="column"
            p={['15px 10px', '20px']}
            mb="20px"
            bg={colorMode === 'light' ? 'mainWhite' : '#111'}
          >
            <Box p="0 0 15px" fontWeight="600" fontSize="28px">
              my bag
            </Box>
            {bag.map((item) => {
              return <BagItem key={`bag-item-${item.id}`} item={item} />
            })}
          </Flex>
          <Flex
            display={['none', null, 'flex']}
            justifyContent="space-between"
            bg={colorMode === 'light' ? 'mainWhite' : '#111'}
            p="20px"
            fontSize="22px"
            fontWeight="600"
          >
            <Box>sub-total</Box>
            <Box>Â£{subtotal}</Box>
          </Flex>
        </Flex>
        <Summary subtotal={subtotal} />
      </Flex>
    </Box>
  )
}

export default Bag
