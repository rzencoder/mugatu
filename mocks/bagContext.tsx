import { bagContext } from '@/context/bagContext'
import { BagItem } from '@/types/bagItem'
import { useState } from 'react'

interface Result {
  title: string
  message: string
  status: string
}

interface BagContextInterface {
  bag: BagItem[]
  fetchBag: () => Promise<void>
  fetchBagOnSignIn: () => Promise<void>
  addToBag: (item: BagItem, signedIn?: boolean) => Promise<Result>
  removeFromBag: (item: BagItem, signedIn?: boolean) => Promise<Result>
}

interface ProvideBagProps {
  bag: BagItem[]
  children: React.ReactNode
}

export const ProvideBag = ({ bag, children }: ProvideBagProps): JSX.Element => {
  const bagValue = useProvideBag(bag)
  return <bagContext.Provider value={bagValue}>{children}</bagContext.Provider>
}

export const useProvideBag = (mockBag: BagItem[]): BagContextInterface => {
  const [bag, setBag] = useState(mockBag)

  const fetchBag = async () => {
    try {
      setBag([...mockBag])
    } catch (error) {
      console.log(error)
    }
  }

  const fetchBagOnSignIn = async () => {
    if (bag.length > 0) {
      // Combine guest shopping bag with users saved bag
      try {
        setBag([...mockBag])
      } catch {
        console.log('there was an error fetching your shopping bag')
      }
    } else {
      try {
        await fetchBag()
      } catch {
        console.log('there was an error fetching your shopping bag')
      }
    }
  }

  const addToBag = async (item: BagItem, signedIn = false) => {
    if (signedIn) {
      try {
        setBag([...bag, item])
        return {
          title: 'Item Added!',
          message: 'Your item has been added to your shopping bag',
          status: 'success',
        }
      } catch {
        return {
          title: 'Error!',
          message: 'There was an error adding your item',
          status: 'error',
        }
      }
    } else {
      const filteredBag = bag.filter((product: BagItem) => product.id !== item.id)
      setBag([...filteredBag, item])
      return {
        title: 'Item Added!',
        message: 'Your item has been added to your shopping bag',
        status: 'success',
      }
    }
  }

  const removeFromBag = async (item: BagItem, signedIn = false) => {
    if (!signedIn) {
      const updatedBag = bag.filter((el: BagItem) => el.id !== item.id)
      setBag(updatedBag)
      return {
        title: 'Item Removed!',
        message: 'Your item has been removed from your shopping bag',
        status: 'success',
      }
    } else {
      try {
        const updatedBag = bag.filter((el: BagItem) => el.id !== item.id)
        setBag(updatedBag)
        return {
          title: 'Item Removed!',
          message: 'Your item has been removed',
          status: 'success',
        }
      } catch {
        return {
          title: 'Error!',
          message: 'There was an error removing your item',
          status: 'error',
        }
      }
    }
  }

  return {
    bag,
    fetchBag,
    fetchBagOnSignIn,
    addToBag,
    removeFromBag,
  } as const
}
