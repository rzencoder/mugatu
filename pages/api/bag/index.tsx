import { addToBag, getBag, removeFromBag } from '@/controllers/bag'
import { NextApiRequest, NextApiResponse } from 'next'

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export default async (req: NextApiRequest, res: NextApiResponse) => {
  const item = req.body
  switch (req.method) {
    case 'GET':
      console.log('here')
      return getBag(req, res)
    case 'POST':
      return addToBag(req, res, item)
    case 'PUT':
      return removeFromBag(req, res, item)
    default:
      //Method not allowed
      return res.status(405).end()
  }
}
