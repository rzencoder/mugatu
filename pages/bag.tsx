import Head from 'next/head'
import { Layout } from '@/components/layouts'
import Bag from '@/components/bag'

export default function BagPage(): JSX.Element {
  return (
    <>
      <Head>
        <title>Shopping Bag | Mugatu</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Layout>
        <Bag />
      </Layout>
    </>
  )
}
