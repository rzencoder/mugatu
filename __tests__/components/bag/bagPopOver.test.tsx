import { fireEvent, render } from '@testing-library/react'
import BagPopOver from '@/components/bag/bagPopOver'
import { ProvideBag } from '@/mocks/bagContext'
import { bag } from '@/mocks/bag'
import { Popover } from '@chakra-ui/popover'

describe('<BagPopOver />', () => {
  it('renders <BagPopOver /> component', () => {
    const { container, getByText } = render(
      <Popover>
        <ProvideBag bag={bag}>
          <BagPopOver />
        </ProvideBag>
      </Popover>
    )
    expect(getByText('2 items')).toBeTruthy()
    expect(getByText('View Bag')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders nothing if no items', () => {
    const { queryByText } = render(
      <Popover>
        <ProvideBag bag={[]}>
          <BagPopOver />
        </ProvideBag>
      </Popover>
    )
    expect(queryByText('View Bag')).toBeNull
  })

  it('removes item is user clicks on delete icon', () => {
    const { queryByText, getByText, getByTestId } = render(
      <Popover>
        <ProvideBag bag={bag}>
          <BagPopOver />
        </ProvideBag>
      </Popover>
    )
    expect(getByText('2 items')).toBeTruthy()
    fireEvent.click(getByTestId('delete-icon-blue-jeans'))
    expect(getByText('1 item')).toBeTruthy()
    expect(queryByText('2 items')).toBeNull
  })
})
