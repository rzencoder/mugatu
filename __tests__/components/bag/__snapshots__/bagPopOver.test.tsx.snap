// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BagPopOver /> renders <BagPopOver /> component 1`] = `
<div
  class="css-0"
  style="visibility: hidden; position: absolute; left: 0px; top: 0px;"
>
  <section
    aria-describedby="popover-body-2"
    aria-labelledby="popover-header-2"
    class="chakra-popover__content css-1hxk7rx"
    id="popover-content-2"
    role="dialog"
    style="transform-origin: top center; opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
    tabindex="-1"
  >
    <button
      aria-label="Close"
      class="css-jss8o7"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
          fill="currentColor"
        />
      </svg>
    </button>
    <header
      class="chakra-popover__header css-11virbg"
      id="popover-header-2"
    >
      my bag
    </header>
    <div
      class="chakra-popover__body css-0"
      id="popover-body-2"
    >
      <div
        class="css-j7qwjs"
      >
        <div
          class="css-35ezg3"
        >
          2 items
        </div>
        <div
          class="css-a0md4"
        >
          <div
            class="css-mvjt7b"
          >
            <div
              class="css-14ubltb"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjE1MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                    style="max-width: 100%; display: block;"
                  />
                </div>
                <img
                  alt="Blue jeans"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            <div
              class="css-2rk6au"
            >
              <div
                class="css-18ugfwx"
              >
                Blue jeans
              </div>
              <div
                class="css-16yexji"
              >
                <div
                  class="css-0"
                >
                  blue
                </div>
                <div
                  class="css-0"
                >
                  size: 
                  6
                </div>
                <div
                  class="css-0"
                >
                  qty: 
                  1
                </div>
              </div>
              <div
                class="css-1lekzkb"
              >
                <div
                  class="css-13ih39a"
                >
                  £
                  29.99
                </div>
                <button
                  class="chakra-button css-oqfhsh"
                  data-testid="delete-icon-blue-jeans"
                  type="button"
                >
                  <svg
                    class="chakra-icon css-qve0ed"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="currentColor"
                    >
                      <path
                        d="M19.452 7.5H4.547a.5.5 0 00-.5.545l1.287 14.136A2 2 0 007.326 24h9.347a2 2 0 001.992-1.819L19.95 8.045a.5.5 0 00-.129-.382.5.5 0 00-.369-.163zm-9.2 13a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zm5 0a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zM22 4h-4.75a.25.25 0 01-.25-.25V2.5A2.5 2.5 0 0014.5 0h-5A2.5 2.5 0 007 2.5v1.25a.25.25 0 01-.25.25H2a1 1 0 000 2h20a1 1 0 000-2zM9 3.75V2.5a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v1.25a.25.25 0 01-.25.25h-5.5A.25.25 0 019 3.75z"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="css-mvjt7b"
          >
            <div
              class="css-14ubltb"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjE1MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                    style="max-width: 100%; display: block;"
                  />
                </div>
                <img
                  alt="Black Shirt"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            <div
              class="css-2rk6au"
            >
              <div
                class="css-18ugfwx"
              >
                Black Shirt
              </div>
              <div
                class="css-16yexji"
              >
                <div
                  class="css-0"
                >
                  black
                </div>
                <div
                  class="css-0"
                >
                  size: 
                  s
                </div>
                <div
                  class="css-0"
                >
                  qty: 
                  1
                </div>
              </div>
              <div
                class="css-1lekzkb"
              >
                <div
                  class="css-13ih39a"
                >
                  £
                  19.99
                </div>
                <button
                  class="chakra-button css-oqfhsh"
                  data-testid="delete-icon-black-shirt"
                  type="button"
                >
                  <svg
                    class="chakra-icon css-qve0ed"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="currentColor"
                    >
                      <path
                        d="M19.452 7.5H4.547a.5.5 0 00-.5.545l1.287 14.136A2 2 0 007.326 24h9.347a2 2 0 001.992-1.819L19.95 8.045a.5.5 0 00-.129-.382.5.5 0 00-.369-.163zm-9.2 13a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zm5 0a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zM22 4h-4.75a.25.25 0 01-.25-.25V2.5A2.5 2.5 0 0014.5 0h-5A2.5 2.5 0 007 2.5v1.25a.25.25 0 01-.25.25H2a1 1 0 000 2h20a1 1 0 000-2zM9 3.75V2.5a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v1.25a.25.25 0 01-.25.25h-5.5A.25.25 0 019 3.75z"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1ythi73"
        >
          <div
            class="css-0"
          >
            subtotal
          </div>
          <div
            class="css-35ezg3"
          >
            £
            49.98
          </div>
        </div>
        <a
          class="chakra-link css-0"
          href="/bag"
        >
          <button
            class="chakra-button css-gjycyk"
            type="button"
          >
            View Bag
          </button>
        </a>
      </div>
    </div>
  </section>
</div>
`;
